R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
as you can see on L240, "master" is passed to `createSparkContext`. If this is not being set as `spark.master`, could you track down what's going on?